app/javascript/shared/helpers/CustomEventHelper.js

```diff
@@ -1,15 +1,15 @@
-export const createEvent = eventName => {
+export const createEvent = ({ eventName, data }) => {
   let event;
   if (typeof window.CustomEvent === 'function') {
-    event = new CustomEvent(eventName);
+    event = new CustomEvent(eventName, { detail: data });
   } else {
     event = document.createEvent('CustomEvent');
     event.initCustomEvent(eventName, false, false, null);
   }
   return event;
 };
 
-export const dispatchWindowEvent = eventName => {
-  const event = createEvent(eventName);
-  window.dispatchEvent(event);
+export const dispatchWindowEvent = ({ eventName, data }) => {
+  const event = createEvent({ eventName, data });
+  window.parent.dispatchEvent(event);
```

Comment:
window.parent shouldn't be accessed directly. This would throw an error if the parent and chatwoot domain is different.